oceanbench/core/lagrangian_trajectory.py

```diff
@@ -222,28 +324,51 @@ def _get_all_particles_positions(
     return all_particle_latitudes, all_particle_longitudes
 
 
-def _get_particle_dataset(dataset: xarray.Dataset, zone: Zone) -> xarray.Dataset:
+def _get_particle_dataset(dataset: xarray.Dataset, latitudes: xarray.Dataset, longitudes: xarray.Dataset) -> xarray.Dataset:
     field_set = _build_field_set(dataset)
-    latitudes, longitudes = _zone_dimensions(dataset, zone)
-    particle_initial_latitudes, particle_initial_longitudes = _particle_initial_positions(latitudes, longitudes)
+    particle_initial_latitudes = latitudes 
+    particle_initial_longitudes = longitudes 
 
-    all_particle_latitudes, all_particle_longitudes = _get_all_particles_positions(
+    ds_out = _get_all_particles_positions(
         dataset,
-        field_set,
         particle_initial_latitudes,
         particle_initial_longitudes,
     )
-    x = all_particle_latitudes.reshape(latitudes.shape[0], longitudes.shape[0], LEAD_DAY_STOP).transpose(2, 0, 1)
-    y = all_particle_longitudes.reshape(latitudes.shape[0], longitudes.shape[0], LEAD_DAY_STOP).transpose(2, 0, 1)
+    return ds_out
 
-    return xarray.Dataset(
-        {
-            "x": (["time", "lat", "lon"], x),
-            "y": (["time", "lat", "lon"], y),
-        },
-        coords={
-            "time": dataset[Dimension.TIME.key()][0:LEAD_DAY_STOP],
-            "lat": latitudes,
-            "lon": longitudes,
-        },
-    )
+
+def get_random_ocean_points_from_file(ds: xarray.Dataset, varname: str = "zos", n: int = 100, seed: int = 42):
+
+    var = ds[varname].isel(lead_day_index=0)  
+    mask = ~numpy.isnan(var)[0].squeeze()
+    lat = ds.lat
+    lon = ds.lon
+
+    lat_grid, lon_grid = numpy.meshgrid(lat, lon, indexing="ij")
+    lat_vals = lat_grid[mask.values]
+    lon_vals = lon_grid[mask.values]
+
+    if len(lat_vals) < n:
+        raise ValueError(f"Requested {n} points, but only {len(lat_vals)} ocean points available.")
+
+    numpy.random.seed(seed)
+    idx = numpy.random.choice(len(lat_vals), n, replace=False)
+
+    return lat_vals[idx], lon_vals[idx]
+
+def Euclidean_distance( modelset: list[xarray.Dataset], refset: list[xarray.Dataset], pad: int = 10) -> list[numpy.ndarray]:
```

Comment:
No caps in python function name.
```suggestion
def euclidean_distance( model_set: list[xarray.Dataset], reference_set: list[xarray.Dataset], padding: int = 10) -> list[numpy.ndarray]:
```